library/src/main/java/com/q42/q42stats/library/Q42StatsPrefs.kt

```diff
@@ -0,0 +1,40 @@
+package com.q42.q42stats.library
+
+import android.content.Context
+import android.content.SharedPreferences
+import java.util.*
+
+private const val SHARED_PREFS_NAME = "Q42StatsPrefs"
+private const val LAST_SUBMIT_TIMESTAMP_KEY = "lastSubmitTimestamp"
+private const val INSTALLATION_ID_KEY = "installationId"
+
+class Q42StatsPrefs(context: Context) {
+    private val prefs: SharedPreferences =
+        context.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE)
+
+    fun getOrCreateInstallationId(): String {
+        val uuid = prefs.getString(INSTALLATION_ID_KEY, null)
+        return uuid ?: createInstallationId()
+    }
+
+    fun withinSubmitInterval(interval: Long) =
+        System.currentTimeMillis() <
+                prefs.getLong(LAST_SUBMIT_TIMESTAMP_KEY, 0) + interval
+
+    fun updateSubmitTimestamp() {
+        prefs.edit().apply() {
```

Comment:
double apply() is a bit confusing. `prefs.edit().putLong(x).apply()` will do.

Reply:
True, To improve readability I used with(prefs.edit()) {} , as suggested [by the docs ](https://developer.android.com/training/data-storage/shared-preferences)